Request/ParamFetcher.php

```diff
@@ -223,10 +222,15 @@ public function cleanParamWithRequirements(Param $config, $param, $strict)
                     $config->requirements
                 ),
             ));
+        } elseif (is_array($config->requirements) && isset($config->requirements["rule"]) && $config->requirements["error_message"]) {
+            $constraint = new Regex(array(
+                'pattern' => '#^'.$config->requirements["rule"].'$#xsu',
+                'message' => $config->requirements["error_message"]
+            ));
+        }
 
-            if (false === $config->allowBlank) {
-                $constraint = array(new NotBlank(), $constraint);
-            }
+        if (false === $config->allowBlank) {
```

Comment:
moving this out of the if introduced a new code path, can you check that this doesn't have any adverse side effects?